src/api/cpp/histogram.cpp

```diff
@@ -26,6 +26,9 @@ array histogram(const array &in, const unsigned nbins, const double minval, cons
 array histogram(const array &in, const unsigned nbins)
 {
     af_array out = 0;
+    if(in.numdims() == 0) {
+        return in;
+    }
```

Comment:
Because the min<T> and max<T> will throw errors with | in | == 0. Checking here let's us return an empty array. 
I did add checks for the pure C API.